pkg/solana/client/classify_errors.go

```diff
@@ -0,0 +1,75 @@
+	ErrAddressLookupTableNotFound            = regexp.MustCompile(`Transaction loads an address table account that doesn't exist`)
+	ErrInvalidAddressLookupTableOwner        = regexp.MustCompile(`Transaction loads an address table account with an invalid owner`)
+	ErrInvalidAddressLookupTableData         = regexp.MustCompile(`Transaction loads an address table account with invalid data`)
+	ErrInvalidAddressLookupTableIndex        = regexp.MustCompile(`Transaction address table lookup uses an invalid index`)
+	ErrInvalidRentPayingAccount              = regexp.MustCompile(`Transaction leaves an account with a lower balance than rent-exempt minimum`)
+	ErrWouldExceedMaxVoteCostLimit           = regexp.MustCompile(`Transaction would exceed max Vote Cost Limit`)
+	ErrWouldExceedAccountDataTotalLimit      = regexp.MustCompile(`Transaction would exceed total account data limit`)
+	ErrDuplicateInstruction                  = regexp.MustCompile(`Transaction contains a duplicate instruction \(\d+\) that is not allowed`)
+	ErrInsufficientFundsForRent              = regexp.MustCompile(`Transaction results in an account \(\d+\) with insufficient funds for rent`)
+	ErrMaxLoadedAccountsDataSizeExceeded     = regexp.MustCompile(`Transaction exceeded max loaded accounts data size cap`)
+	ErrInvalidLoadedAccountsDataSizeLimit    = regexp.MustCompile(`LoadedAccountsDataSizeLimit set for transaction must be greater than 0\.`)
+	ErrResanitizationNeeded                  = regexp.MustCompile(`Sanitized transaction differed before/after feature activation\. Needs to be resanitized\.`)
+	ErrProgramExecutionTemporarilyRestricted = regexp.MustCompile(`Execution of the program referenced by account at index \d+ is temporarily restricted\.`)
+	ErrUnbalancedTransaction                 = regexp.MustCompile(`Sum of account balances before and after transaction do not match`)
+	ErrProgramCacheHitMaxLimit               = regexp.MustCompile(`Program cache hit max limit`)
+)
+
+// errCodes maps regex patterns to their corresponding return code
+// errors are considered Retryable by default if not in this map
+var errCodes = map[*regexp.Regexp]mn.SendTxReturnCode{
+	ErrSanitizeFailure:         mn.Fatal,                   // Transaction formatting is invalid and cannot be processed or retried
+	ErrAlreadyProcessed:        mn.TransactionAlreadyKnown, // Transaction was already processed and thus known by the RPC
+	ErrInsufficientFundsForFee: mn.InsufficientFunds,       // Transaction was rejected due to insufficient funds for gas fees
+}
+
+// ClassifySendError returns the corresponding return code based on the error.
+func ClassifySendError(_ *solana.Transaction, err error) mn.SendTxReturnCode {
+	if err == nil {
+		return mn.Successful
+	}
+
+	errMsg := err.Error()
+	for pattern, code := range errCodes {
+		if pattern.MatchString(errMsg) {
+			return code
+		}
+	}
+	return mn.Retryable
```

Comment:
Eventually, we want this to be unknown. It will help us spot misconfigured RPC more easily. And ensure we properly handle transactions. It does not seem to be a blocker for the launch. So lets just create a ticket to track that effort.

Reply:
Created a ticket: https://smartcontract-it.atlassian.net/browse/NONEVM-796